src/tools/schema2kotlin.ts

```diff
@@ -65,23 +69,27 @@ import arcs.StringDecoder
 
 class KotlinGenerator implements ClassGenerator {
   fields: string[] = [];
+  fieldVals: string[] = [];
+  fieldSets: string[] = [];
   encode: string[] = [];
   decode: string[] = [];
 
   constructor(readonly node: SchemaNode) {}
 
   addField(field: string, typeChar: string) {
-    const {type, decodeFn} = typeMap[typeChar];
+    const {type, decodeFn, delegate} = typeMap[typeChar];
     const fixed = field + (keywords.includes(field) ? '_' : '');
 
-    this.fields.push(`var ${fixed}: ${type}`);
+    this.fields.push(`${fixed}: ${type}`);
+    this.fieldVals.push(`var ${fixed} by ${delegate}`);
+    this.fieldSets.push(`this.${fixed} = ${fixed}`);
 
     this.decode.push(`"${field}" -> {`,
-                     `  decoder.validate("${typeChar}")`,
-                     `  this.${fixed} = decoder.${decodeFn}`,
+                     `    decoder.validate("${typeChar}")`,
+                     `    this.${fixed} = decoder.${decodeFn}`,
```

Comment:
It's hard to tell in the PR that the generated code will have the correct whitespace. Can you confirm?

Reply:
Looking below, it seems like the whitespace has changed, so 👍

Reply:
Unrelated to this PR: we should check in an example of the generated code as a reference (like src/wasm/cpp/entity-class.h).

Reply:
Agreed, and I've triple checked the whitespace is correct.